classes/local/store/object_file_system.php

```diff
@@ -751,11 +756,62 @@ public function presigned_url_should_redirect($contenthash) {
      * @param array $headers request headers.
      *
      * @return string.
+     * @throws \dml_exception
      */
     public function generate_presigned_url_to_external_file($contenthash, $headers = array()) {
+        if ($this->is_white_listed($contenthash)) {
+            return $this->get_url_by_contenthash($contenthash);
```

Comment:
isn't this dead code? if the file type is black listed then we should not get here, if we are white listed then the signed url shouldn't change?